### PR TITLE
Feature/tshaiman/terraform sample

### PR DESCRIPTION
**Pull request by tshaiman**
- Adding Terraform samples and tutorial for complete End-2-End Media Solution with dependent resource
- Change folder names

### CODE REVIEW

File: ARM-Quick-Start/README.md [hunk 1]
```diff
@@ -19,14 +19,16 @@ The resource will be deployed to your subscription and will create the Azure Vid
 
 * An Azure Media Services (AMS) account. You can create one for free through the [Create AMS Account](https://docs.microsoft.com/en-us/azure/media-services/latest/account-create-how-to).
 
+* In case you are interested in creating the Full End-To-End Media Solution that Creates Azure Media Service Accounnt, Storage Account with all the permissions correctly wired, please chaeck the [Create Media Solution - Terraform](../Create-Media-Solution-Terraform/) Demo.
```
Comment (tshoham) [inline ARM-Quick-Start/README.md:22]: small typos-
Account*, check*

File: ApiUsage/ArmBased/README.md [hunk 2]
```diff
@@ -1,25 +1,25 @@
 ---
 page_type: How to use AVAM API
 languages:
-- C#
-- .Net
+  - C#
+  - .Net
 products:
-- azure-video-analyzer-for-media
+  - azure-video-analyzer-for-media
```
Comment (tshoham) [inline ApiUsage/ArmBased/README.md:7]: Should the name be changed to VI? (maybe this is a task to update all our samples)

File: Terraform-Deploy/README.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+```powershell
+terraform init
+```
+in order to install required dependencies and libraries.
+
+2. Next run the `plan` command in order to view the changes that are about to be deployed to Azure.
+
+```powershell
+terraform init
```
Comment (tshoham) [inline Terraform-Deploy/README.md:59]: Should this be 'plan'?

**Review by tshoham — APPROVED** [review]
This looks awesome!
Left a few small comments

File: ApiUsage/ArmBased/README.md (hunk 2 above)
Comment (Itnorman) [reply ApiUsage/ArmBased/README.md:7]: Yes we should!

Comment (tshaiman) [reply ApiUsage/ArmBased/README.md:7]: what is the official new name ? 
Azure Video Indexer ? 
Video Indexer  ?

**Pull request merged**